### ISSUE TITLE
AzureSearchableSnapshotsIT#testSearchResultsWhenFrozen timed out

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#58260]
**Build scan**:
[scan](https://gradle-enterprise.elastic.co/s/dq5lehlmavc56/tests/:x-pack:plugin:searchable-snapshots:qa:azure:integTestRunner/org.elasticsearch.xpack.searchablesnapshots.AzureSearchableSnapshotsIT/testSearchResultsWhenFrozen#1)

**Repro line**:
```
./gradlew ':x-pack:plugin:searchable-snapshots:qa:azure:integTestRunner' --tests "org.elasticsearch.xpack.searchablesnapshots.AzureSearchableSnapshotsIT.testSearchResultsWhenFrozen" -Dtests.seed=E9C97AC877AA4049 -Dtests.security.manager=true -Dtests.locale=ar-SA -Dtests.timezone=Europe/Kiev -Druntime.java=11
```

**Reproduces locally?**:
Not for me, but I likely don't have the right setup to run it properly.

**Applicable branches**: 7.x and master

**Failure history**:
https://build-stats.elastic.co/goto/3ef7c8c9ff2a663f17d0ec94ad117eb2

**Failure excerpt**:
```
Caused by: java.net.SocketTimeoutException: 60,000 milliseconds timeout on connection http-outgoing-2 [ACTIVE]
```

**Comment by dakrone** [comment]
A failure similar to this keeps happening on my 7.x branch, the failure from Jenkins is:

```
13:00:22 org.elasticsearch.xpack.searchablesnapshots.AzureSearchableSnapshotsIT > testStats FAILED
13:00:22     java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-2 [ACTIVE]
13:00:22         at __randomizedtesting.SeedInfo.seed([62E41241059D1483:F8654BFE6D41DAA0]:0)
13:00:22         at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:818)
13:00:22         at org.elasticsearch.client.RestClient.performRequest(RestClient.java:250)
13:00:22         at org.elasticsearch.client.RestClient.performRequest(RestClient.java:253)
13:00:22         at org.elasticsearch.client.RestClient.performRequest(RestClient.java:237)
13:00:22         at org.elasticsearch.xpack.searchablesnapshots.AbstractSearchableSnapshotsRestTestCase.registerRepository(AbstractSearchableSnapshotsRestTestCase.java:255)
13:00:22         at org.elasticsearch.xpack.searchablesnapshots.AbstractSearchableSnapshotsRestTestCase.runSearchableSnapshotsTest(AbstractSearchableSnapshotsRestTestCase.java:56)
13:00:22         at org.elasticsearch.xpack.searchablesnapshots.AbstractSearchableSnapshotsRestTestCase.testStats(AbstractSearchableSnapshotsRestTestCase.java:164)
```

And in the ES logs there are the following errors:

```
[2020-06-30T19:00:24,576][WARN ][r.suppressed             ] [integTest-0] path: _snapshot/repository, params: {repository=repository}
org.elasticsearch.repositories.RepositoryVerificationException: [repository] path [_searchable_snapshots_tests_62E41241059D1483] is not accessible on master node
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1165) ~[elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	at org.elasticsearch.repositories.RepositoriesService$3.doRun(RepositoriesService.java:248) ~[elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:710) [elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_241]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_241]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_241]
Caused by: java.io.IOException: Can not write blob master.dat
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:116) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:122) ~[?:?]
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1160) ~[elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	... 6 more
Caused by: com.microsoft.azure.storage.StorageException: An unknown failure occurred : Unexpected end of file from server
	at com.microsoft.azure.storage.StorageException.translateException(StorageException.java:67) ~[?:?]
	at com.microsoft.azure.storage.core.ExecutionEngine.executeWithRetry(ExecutionEngine.java:220) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.uploadFullBlob(CloudBlockBlob.java:1035) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.upload(CloudBlockBlob.java:864) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.upload(CloudBlockBlob.java:743) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobStore.lambda$writeBlob$18(AzureBlobStore.java:323) ~[?:?]
	at org.elasticsearch.repositories.azure.SocketAccess.lambda$doPrivilegedVoidException$0(SocketAccess.java:69) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_241]
	at org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedVoidException(SocketAccess.java:68) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobStore.writeBlob(AzureBlobStore.java:322) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:114) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:122) ~[?:?]
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1160) ~[elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	... 6 more
Caused by: java.net.SocketException: Unexpected end of file from server
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:851) ~[?:?]
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:678) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1593) ~[?:?]
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498) ~[?:?]
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480) ~[?:1.8.0_241]
	at com.microsoft.azure.storage.core.ExecutionEngine.executeWithRetry(ExecutionEngine.java:115) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.uploadFullBlob(CloudBlockBlob.java:1035) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.upload(CloudBlockBlob.java:864) ~[?:?]
	at com.microsoft.azure.storage.blob.CloudBlockBlob.upload(CloudBlockBlob.java:743) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobStore.lambda$writeBlob$18(AzureBlobStore.java:323) ~[?:?]
	at org.elasticsearch.repositories.azure.SocketAccess.lambda$doPrivilegedVoidException$0(SocketAccess.java:69) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_241]
	at org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedVoidException(SocketAccess.java:68) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobStore.writeBlob(AzureBlobStore.java:322) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlob(AzureBlobContainer.java:114) ~[?:?]
	at org.elasticsearch.repositories.azure.AzureBlobContainer.writeBlobAtomic(AzureBlobContainer.java:122) ~[?:?]
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:1160) ~[elasticsearch-7.9.0-SNAPSHOT.jar:7.9.0-SNAPSHOT]
	... 6 more
```

I haven't been able to reproduce it locally however (although it fails every time on my 7.x PR). A build scan is at https://gradle-enterprise.elastic.co/s/zrslbr2ihbsvs

**Comment by dakrone** [comment]
I muted this on 7.x

**Comment by breskeby** [comment]
I cherry-picked the mute down to 7.8

**Comment by tlrx** [comment]
The test initially failed because of repository base paths being the same for tests running in parallel: this has been fixed in #58664. Then the test times out because of a deadlock that is now fixed by #59595.

With these two changes now merged, I'm going to unmute the test on master and see how it goes. If there's no failures then I'll unmute in all branches. 

Please do reopen if the test fails again on master.

**Comment by tlrx** [comment]
This test has been unmuted on 7.x (#59749), 7.9 (#59752) and 7.8 (#59750).
